### PR TITLE
[TextField] Make autoComplete prop a required string

### PR DESCRIPTION
**Pull request by andrewmcgov**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/core-issues/issues/26319

Makes `autoComplete` a required prop on the `TextField` component as part of the [autocomplete project](https://vault.shopify.io/projects/17459).

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';

import {Page, TextField} from '../src';

export function Playground() {
  const [email, setEmail] = React.useState('');
  return (
    <Page title="Playground">
      <TextField
        label="Email"
        type="email"
        value={email}
        onChange={setEmail}
        autoComplete="email"
      />
    </Page>
  );
}

```

</details>

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

**Review by alex-page — APPROVED** [review]

**Comment by alex-page** [comment]
This will require a changelog with some instructions on what do update in the breaking change.

File: src/components/Filters/Filters.tsx [hunk 1]
```diff
@@ -287,6 +287,7 @@ class FiltersInner extends Component<CombinedProps, State> {
             clearButton
             onClearButtonClick={onQueryClear}
             disabled={disabled}
+            autoComplete="off"
```
Comment (brilliant-ember) [inline src/components/Filters/Filters.tsx:290]: every where else in this PR autoComplete="nope" but here it is "off" not sure if this is important but just wanted to bring it up.

[the remark this answers is not in the document]
Comment (andrewmcgov) [reply src/components/Filters/Filters.tsx:290]: Good catch thanks!

File: src/components/TextField/tests/TextField.test.tsx [hunk 2]
```diff
@@ -179,43 +217,49 @@ describe('<TextField />', () => {
...
-    it('sets autoComplete to string value when string is given', () => {
-      const textField = mountWithAppProvider(
-        <TextField
-          label="TextField"
-          autoComplete="firstName"
-          onChange={noop}
-        />,
-      );
-      expect(textField.find('input').prop('autoComplete')).toBe('firstName');
+      expect(textField.find('input').prop('autoComplete')).toBe('nope');
```
Comment (alex-page) [inline src/components/TextField/tests/TextField.test.tsx:250]: Awesome 💯

**Review by alex-page — APPROVED** [review]

File: src/components/TextField/TextField.tsx [hunk 3]
```diff
@@ -522,11 +522,9 @@ export function TextField({
   }
 }
 
-function normalizeAutoComplete(autoComplete?: boolean | string) {
-  if (autoComplete === true) {
-    return 'on';
-  } else if (autoComplete === false) {
-    return 'off';
+function normalizeAutoComplete(autoComplete: string, name?: string) {
+  if (autoComplete === 'off' && !name) {
```
Comment (gcornhill) [inline src/components/TextField/TextField.tsx:526]: I think the logic is if === off && there is a name --> Change to nope.
If no name and off, leave it alone.

**Review by gcornhill — APPROVED** [review]

**Pull request merged**